lib/build.js

```diff
@@ -99,7 +102,7 @@ class BuildModel extends BaseModel {
 
         delete this.user;
         const factory = UserFactory.getInstance();
-        const user = factory.get({ username: this.username });
+        const user = factory.get({ username: this.username, scmContext: this.scmContext });
```

Comment:
It turned out that we don't use this function at all. Guess we can just remove this and there is no need to pass the `scmContext` in to constructor. Can you double check to see if we use this function any where?